tests/api/runtimes/base.py

```diff
@@ -0,0 +1,207 @@
+            )
+            container_status = client.V1ContainerStatus(
+                state=container_state,
+                image=self.image_name,
+                image_id="must-provide-image-id",
+                name="must-provide-name",
+                ready=True,
+                restart_count=0,
+            )
+            status = client.V1PodStatus(
+                phase=PodPhases.succeeded, container_statuses=[container_status]
+            )
+            response_pod = deepcopy(pod)
+            response_pod.status = status
+            response_pod.metadata.name = "test-pod"
+            response_pod.metadata.namespace = namespace
+            return response_pod
+
+        get_k8s().v1api.create_namespaced_pod = unittest.mock.Mock(
+            side_effect=_generate_pod
+        )
+
+    def _execute_run(self, runtime, **kwargs):
+        # Reset the mock, so that when checking is create_pod was called, no leftovers are there (in case running
+        # multiple runs in the same test)
+        get_k8s().v1api.create_namespaced_pod.reset_mock()
+
+        runtime.run(
+            name=self.name,
+            project=self.project,
+            artifact_path=self.artifact_path,
+            **kwargs,
+        )
+
+    def _assert_labels_exist_in_pod_creation(self, labels: dict):
+        expected_labels = {
+            "mlrun/class": "job",
+            "mlrun/name": self.name,
+            "mlrun/project": self.project,
+            "mlrun/tag": "latest",
+        }
+
+        for key in expected_labels:
+            assert labels[key] == expected_labels[key]
+
+    def _assert_function_config_as_expected(
+        self,
+        config,
+        expected_params,
+        expected_inputs,
+        expected_hyper_params,
+        expected_secrets,
+    ):
+        function_metadata = config["metadata"]
+        assert function_metadata["name"] == self.name
+        assert function_metadata["project"] == self.project
+
+        function_spec = config["spec"]
+        assert function_spec["output_path"] == self.artifact_path
+        if expected_params:
+            assert function_spec["parameters"] == expected_params
+        if expected_inputs:
+            assert function_spec["inputs"] == expected_inputs
+        if expected_hyper_params:
+            assert function_spec["hyperparams"] == expected_hyper_params
+        if expected_secrets:
+            assert function_spec["secret_sources"] == [expected_secrets]
+
+    @staticmethod
+    def _assert_env_variables_set_correctly_in_pod_env(pod_env, expected_variables):
+        for env_variable in pod_env:
+            name = env_variable["name"]
+            if name in expected_variables:
+                if expected_variables[name]:
+                    assert expected_variables[name] == env_variable["value"]
+                expected_variables.pop(name)
+
+        # Make sure all variables were accounted for
+        assert len(expected_variables) == 0
```

Comment:
Currently if pod env has an env var that the expected doesn't have it will pass, do we really need this ? can't we do full assertion on the env ?

Reply:
I don't want to enforce the exact list of env variables, since some of them may be unrelated to my testing, and I don't want to start checking each and every one of the env variables. For now I'm just making sure that everything I'm looking for is there, and let extras remain there.